hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/SnapshotDiffJob.java

```diff
@@ -316,12 +316,17 @@ public Class<SnapshotDiffJob> getTypeClass() {
 
     @Override
     public byte[] toPersistedFormatImpl(SnapshotDiffJob object) throws IOException {
-      return MAPPER.writeValueAsBytes(object);
+      return object.toProtoBuf().toByteArray();
```

Comment:
@yuxian0908 , thanks for working on this!

We should add a test to verify if the old and the new implementations return the same bytes; see `TestPipelineIDCodec` and `OldPipelineIDCodecForTesting` as example.